controllers/core/flagsourceconfiguration/controller_test.go

```diff
@@ -0,0 +1,170 @@
+	const (
+		testNamespace  = "test-namespace"
+		fsConfigName   = "test-config"
+		deploymentName = "test-deploy"
+	)
+
+	tests := []struct {
+		name       string
+		fsConfig   *v1alpha1.FlagSourceConfiguration
+		deployment *appsv1.Deployment
+		restarted1 string
+		restarted2 string
+	}{
+		{
+			name:       "deployment gets restarted with rollout",
+			fsConfig:   createTestFSConfig(fsConfigName, testNamespace, deploymentName, true),
+			deployment: createTestDeployment(fsConfigName, testNamespace, deploymentName),
+			restarted1: "",
+			restarted2: time.Now().Format(time.RFC3339),
+		},
+		{
+			name:       "deployment without rollout",
+			fsConfig:   createTestFSConfig(fsConfigName, testNamespace, deploymentName, false),
+			deployment: createTestDeployment(fsConfigName, testNamespace, deploymentName),
+			restarted1: "",
+			restarted2: "",
+		},
+		{
+			name:       "no deployment",
+			fsConfig:   createTestFSConfig(fsConfigName, testNamespace, deploymentName, true),
+			deployment: nil,
+			restarted1: "",
+			restarted2: "",
+		},
+	}
+
+	err := v1alpha1.AddToScheme(scheme.Scheme)
+	require.Nil(t, err)
+
+	req := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      fsConfigName,
+		},
+	}
+
+	ctx := context.TODO()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var fakeClient client.Client
+			if tt.deployment != nil {
+				fakeClient = fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(tt.fsConfig, tt.deployment).WithIndex(&appsv1.Deployment{}, fmt.Sprintf("%s/%s", common.OpenFeatureAnnotationPath, common.FlagSourceConfigurationAnnotation), common.FlagSourceConfigurationIndex).Build()
+			} else {
+				fakeClient = fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(tt.fsConfig).WithIndex(&appsv1.Deployment{}, fmt.Sprintf("%s/%s", common.OpenFeatureAnnotationPath, common.FlagSourceConfigurationAnnotation), common.FlagSourceConfigurationIndex).Build()
+			}
+
+			r := &FlagSourceConfigurationReconciler{
+				Client: fakeClient,
+				Log:    ctrl.Log.WithName("flagsourceconfiguration-controller"),
+				Scheme: fakeClient.Scheme(),
+			}
+
+			if tt.deployment != nil {
+				deployment := &appsv1.Deployment{}
+				err = fakeClient.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: testNamespace}, deployment)
+				require.Nil(t, err)
+				restartAt := deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"]
+				require.Equal(t, tt.restarted1, restartAt)
+			}
+
+			_, err = r.Reconcile(ctx, req)
+			require.Nil(t, err)
+
+			if tt.deployment != nil {
+				deployment := &appsv1.Deployment{}
+				err = fakeClient.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: testNamespace}, deployment)
+				require.Nil(t, err)
+
+				require.Equal(t, tt.restarted2, deployment.Spec.Template.ObjectMeta.Annotations["kubectl.kubernetes.io/restartedAt"])
```

Comment:
Is there a chance that this assertion fails if things are really slow or these tests are reordered? Might it be better to instead test that `kubectl.kubernetes.io/restartedAt` is greater than `restarted2`? Maybe I'm confused abou twhat this test is doing.

Reply:
The tests can be reordered and everything should work as expected. Each Iteration is completely independent from the other as we are using `t.Run()`, which is basically a new test with clean context.

 If the test is slow, hard to say to be honest, maybe worth merging and if we have some problems or flakyness in the future, we can always adapt it. Would this work ?

Reply:
sure, would better description/comments do the trick for you ?

Reply:
@odubajDT  IMO yes. May be comment explaining what we do here and better naming for `restarted1` & `restarted2`

Reply:
fixed :+1:

Reply:
@toddbaert I think this is fixed. Can you also check ? :)